### PR TITLE
Add cohort description backend

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

https://github.com/PostHog/posthog/pull/4574 added cohort description as a feature to the frontend… but not to the backend.
This makes description work.
Zrzut ekranu 2021-09-23 o 18 07 15
Reported by user on Slack: https://posthogusers.slack.com/archives/C01FKPGG5U6/p1632410656096000

## Checklist

- [x] Django backend tests

### CODE REVIEW

File: posthog/models/cohort.py [hunk 1]
```diff
@@ -71,6 +71,7 @@ def create(self, *args: Any, **kwargs: Any):
 
 class Cohort(models.Model):
     name: models.CharField = models.CharField(max_length=400, null=True, blank=True)
+    description: models.CharField = models.CharField(max_length=400, blank=True)
```
Comment (neilkakkar) [inline posthog/models/cohort.py:74]: Same constraint on description size as name size sounds too restrictive?

Comment (Twixes) [reply posthog/models/cohort.py]: Well, 400 seems like probably too long for a cohort name and should be enough for most descriptions, but alright, increased the limit here to 1000

Comment (neilkakkar) [reply posthog/models/cohort.py]: That makes sense too! Don't mind going the reverse way, assuming no one has that long cohort names 😋

**Review by neilkakkar — APPROVED** [review]

**Comment by timgl** [comment]
🚓  EE Police. Should this be an EE feature? It's very much a collaboration feature and on similar playing field to the events tags/descriptions we added recently.

**Comment by Twixes** [comment]
I think it can be EE and I get the reasoning, but at the same time it feels awkward to make the description EE by itself. Like, I can use everything, but when I click the description field, THAT is when I see a "upgrade to gain access to this" modal?
I suppose we can also just hide the field entirely if the premium plan isn't available.
Thoughts on which way to go?

**Comment by timgl** [comment]
>  Like, I can use everything, but when I click the description field, THAT is when I see a "upgrade to gain access to this" modal?

For events/property descriptions, we just hide it completely right now which isn't ideal, but the point is we can only ever move in one direction (making something free that was paid) but we can never go the other way. This is a classic feature that falls under collaboration so this needs to be paid and in EE.

**Comment by Twixes** [comment]
Very well, the text box is now only shown if the organization has premium feature `DASHBOARD_COLLABORATION` available.

**Comment by paolodamico** [comment]
To clarify @timgl, is it fine to have it frontend guarded/hidden, but have the backend functionality in the main model? For context, we did separate models for events/properties taxonomy but they were significantly larger features while this is just a simple description.

**Review by paolodamico — APPROVED** [review]

**Pull request merged**